fc/code_generation.py

```diff
@@ -131,7 +135,39 @@ def uname(name):
     return symbols
 
 
-def create_weblab_model(path, class_name, model, outputs, parameters):
+_ONTOLOGY = None
+
+
+def get_variables_transitively(model, term):
+    global _ONTOLOGY
+
+    if _ONTOLOGY is None:
+        # Load oxmeta ontology
+        g = _ONTOLOGY = rdflib.Graph()
+        oxmeta_ttl = pkg_resources.resource_stream('fc', 'ontologies/oxford-metadata.ttl')
+        g.parse(oxmeta_ttl, format='turtle')
+
+    term = create_rdf_node(term)
+    pred_is = create_rdf_node(('http://biomodels.net/biology-qualifiers/', 'is'))
+    pred_is_ver = create_rdf_node(('http://biomodels.net/biology-qualifiers/', 'isVersionOf'))
+
+    cmeta_ids = set()
+    for annotation in _ONTOLOGY.transitive_subjects(rdflib.RDF.type, term):
+        cmeta_ids.update(model.rdf.subjects(pred_is, annotation))
+        cmeta_ids.update(model.rdf.subjects(pred_is_ver, annotation))
+    symbols = []
+    for cmeta_id in cmeta_ids:
+        assert isinstance(cmeta_id, rdflib.URIRef), 'Non-resource {} annotated.'.format(cmeta_id)
+        cmeta_id = str(cmeta_id)
+        if cmeta_id[0] != '#':
+            # TODO This should eventually be implemented
+            raise NotImplementedError(
+                'Non-local annotations are not supported.')
+        symbols.append(model.get_symbol_by_cmeta_id(cmeta_id[1:]))
```

Comment:
Should we not be storing the info about how we found these variables somewhere? Or is it cheap enough to do that in a separate pass?